### PR TITLE
ci: move benchmark stage to run in a tool Docker container

### PR DESCRIPTION
**Pull request by kuisathaverat**
move benchmark stage to run in a tool Docker container

related to https://github.com/elastic/apm-server/pull/2268

### CODE REVIEW

File: script/jenkins/bench.sh [hunk 1]
```diff
@@ -1,9 +1,4 @@
 #!/usr/bin/env bash
 set -euxo pipefail
 
-source ./_beats/dev-tools/common.bash
-
-jenkins_setup
-
-go get -u golang.org/x/tools/cmd/benchcmp
-make bench | tee bench.out
+go test -benchmem -run=XXX -benchtime=100ms -bench='.*' ./... | tee bench.out
```
Comment (kuisathaverat) [inline script/jenkins/bench.sh]:
```suggestion
make bench | tee bench.out
```

[the remark this answers is not in the document]
Comment (kuisathaverat) [reply script/jenkins/bench.sh]: I do not have a really good reason yet, let's keep it as it and remove only the tooling part

File: script/jenkins/bench.sh [hunk 2]
```diff
@@ -1,9 +1,4 @@
 #!/usr/bin/env bash
 set -euxo pipefail
 
-source ./_beats/dev-tools/common.bash
-
-jenkins_setup
-
-go get -u golang.org/x/tools/cmd/benchcmp
```
Comment (simitt) [inline script/jenkins/bench.sh]: Where is this vendored/installed now?

Comment (kuisathaverat) [reply script/jenkins/bench.sh]: the command runs in a Docker container with all the tools installed

**Pull request merged**